src/app/project/project-permissions/project-permissions.component.ts

```diff
@@ -67,30 +69,61 @@ export class ProjectPermissionsComponent implements OnInit {
       .subscribe(params => {
         this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params);
       });
+      this.getUsers();
+  }
 
+  /**
+   * Retrieve the user list, with permissions info, from the user service
+   * 
+   * @return {void}
+   */
+  getUsers(): void {
     this.userService.getAll()
       .toPromise()
       .then((user: User) => {
-        this.users = get(user, 'data');
-        this.tableParams.totalListItems = get(user, 'totalCount');
-        //return user;
-        this.setRowData();
-        this.loading = false;
+        this.userVault = get(user, 'data');
+        this.removeDuplicateUsers();
+        this.tableParams.totalListItems = this.userVault.length || 0;
+        this.paginateUsers(this.tableParams.currentPage || 1);
       })
       .catch(error => {
         console.error(error);
       });
   }
 
+  /**
+   * Remove users that have duplicate names, favouring entries that have permissions.
+   * 
+   * @return {void}
+   */
+  removeDuplicateUsers(): void {
```

Comment:
It seems like this version of `removeDuplicateUsers` is nearly identical to the other, except in how it handles a couple of variables. Did you look at trying make this a shared utility function?

Reply:
Yes I was trying to figure this out... on line 157 I assign the promise return to this.entriesVault. For some reason, the type of the promise return is User (even though it's an array of users). I tried changing the typing down the chain, but then it caused issues in the API file. I will look into it further and try to resolve it with stronger typing.

Reply:
If it's ok with you, I'd like to make an optimization ticket where I streamline the usage of sort, paginate, and removeDuplicate functions throughout the app. I think I could save us at least 50 lines of code.